### PR TITLE
Rhornung67/doc updates

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR contains an update to the User Guide in preparation for the next RAJA release.
- It adds a description of the HIP execution policies, plus numerous small modifications from proofreading several sections, including text cleanup, formatting fixes, fixes to broken code example labels, etc.

Here is a link to the generated docs from this PR: https://raja.readthedocs.io/en/rhornung67-doc-updates/

### CODE REVIEW

**Review by mdavis36 — DISMISSED** [review]

File: docs/sphinx/user_guide/feature/sort.rst [hunk 1]
```diff
@@ -80,57 +82,59 @@ RAJA unstable sort operations look like the following:
  * ``RAJA::sort< exec_policy >(iter, iter + N)``
  * ``RAJA::sort< exec_policy >(iter, iter + N, comparator)``
 
-For example sorting the ``in`` array filled with this sequence of values::
+For example, sorting an array with this sequence of values::
 
    6 7 2 1 0 9 4 8 5 3 4 9 6 3 7 0 1 8 2 5
 
-by performing a sequential unstable sort operation using the following code:
+with an sequential unstable sort operation:
```
Comment (artv3) [inline docs/sphinx/user_guide/feature/sort.rst:89]:
```suggestion
with a sequential unstable sort operation:
```

**Review by artv3 — APPROVED** [review]

File: docs/sphinx/user_guide/feature/view.rst [hunk 2]
```diff
@@ -85,49 +86,60 @@ access array entries with stride N :subscript:`n` * N :subscript:`(n-1)` * ... *
 MultiView
 ^^^^^^^^^^^^^^^^
 
+When using numerous arrays with the same size and Layout, where each needs
```
Comment (rchen20) [inline docs/sphinx/user_guide/feature/view.rst:89]:
```suggestion
Using numerous arrays with the same size and Layout, where each needs 

```

**Review by MrBurmark — APPROVED** [review]

File: docs/sphinx/user_guide/feature/view.rst [hunk 3]
```diff
@@ -85,49 +86,60 @@ access array entries with stride N :subscript:`n` * N :subscript:`(n-1)` * ... *
...
+with a ``RAJA::MultiView`` with the array-of-pointers index set to the 2nd 
+position.
  
 .. literalinclude:: ../../../../examples/multiview.cpp
    :start-after: _multiview_example_2Daopindex_start
    :end-before: _multiview_example_2Daopindex_end
    :language: C++
 
-.. note:: MultiView has not yet been tested with atomic accesses. 
+.. note:: ``RAJA::MultiView`` should work with atomic accesses, but we have not yet developed comprehensive tests for this.
```
Comment (rchen20) [inline docs/sphinx/user_guide/feature/view.rst:142]: We can remove this if this PR https://github.com/LLNL/RAJA/pull/917 can be squeezed in.

File: docs/sphinx/user_guide/feature/sort.rst (hunk 1 above)
Comment (rhornung67) [reply docs/sphinx/user_guide/feature/sort.rst:89]: Fixed.

File: docs/sphinx/user_guide/feature/view.rst (hunk 2 above)
Comment (rhornung67) [reply docs/sphinx/user_guide/feature/view.rst:89]: Fixed.

File: docs/sphinx/user_guide/feature/view.rst (hunk 3 above)
Comment (rhornung67) [reply docs/sphinx/user_guide/feature/view.rst:142]: Removed.

**Comment by rhornung67** [comment]
@MrBurmark @artv3 @mdavis36 @artv3 I've addressed everyone's comments, which dismissed the approvals. Please re-approve when you have a minute.  Thanks.

**Review by mdavis36 — APPROVED** [review]

**Pull request merged**